src/runtime/metal.cpp

```diff
@@ -717,10 +718,17 @@ WEAK int halide_metal_run(void *user_context,
             }
         }
 
-        if (total_args_size < 4096 && metal_api_supports_set_bytes) {
+        // The Metal compiler introduces padding up to a multiple of 4 bytes
+        // in the struct, per email communication from Apple
+        size_t padded_args_size = (total_args_size + 4 - 1) & ~(4 - 1);
```

Comment:
I'd like to change this to "... & ~((size_t)(4 - 1));" (What happens currently on systems with 32-bit ints and 64-bit pointers is the expression produces 0xfffffffc and that gets sign extended to the 64-bit value, but this depends on twos complement, etc. Better to make both the use of unsigned and "same size as a pointer" clear.)

Reply:
Good catch-- fixed, now.